### DISCUSSION TITLE
Preserve images when copypasting from sites

### DISCUSSION THREAD

**Discussion by norstbox** [outline/outline#2748] [Ideas]
Consider the situation: copypasting content with image(s) from some site webpage to the Outline page.

**Current behavior**
Pasted image `src` attribute refers to the original source.

**Proposed behavior**
Store image in the Outline S3 storage. Pasted image `src` attribute refers to the copy of the original image in our storage.

**Rationale for proposed behavior**
Original source may become unavailable at any time for a variety of reasons. Links may become invalid after changes in site structure. Article page (and associated images) may be removed. After all, web site may be down for good.

Proposed behavior will preserve image on our Outline page if any of the abovementioned events occur.

**Comment by tommoor** [comment]
Yes, this is a good idea :)

**Reply by norstbox** [reply]
I'm sure that this seemingly small feature is important for knowledge base oriented software - details make all the difference. Outline loaded with nice features and this will be a good addition :)
